src/core/shape_inference/include/extract_image_patches_shape_inference.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#pragma once
+#include <openvino/op/extractimagepatches.hpp>
+
+namespace ov {
+namespace op {
+namespace v3 {
+template <class T>
+void shape_infer(const ExtractImagePatches* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 1 && output_shapes.size() == 1);
+    const auto& input_shape = input_shapes[0];
+    auto& output_shape = output_shapes[0];
+    output_shape.resize(4);
+    NODE_VALIDATION_CHECK(op, input_shape.rank() == 4, "input tensor must be 4D tensor.");
```

Comment:
Rank might be dynamic. Please take this into account in further logic
```suggestion
    NODE_VALIDATION_CHECK(op, input_shape.rank().compatible(4), "input tensor must be 4D tensor.");
```